src/cli/allow_ip.rs

```diff
@@ -0,0 +1,147 @@
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        allow_ip(self.state.clone(), engine_state, stack, call, input)
+    }
+}
+
+fn allow_ip(
+    state: Arc<Mutex<State>>,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    let span = call.head;
+    let ctrl_c = engine_state.ctrlc.as_ref().unwrap().clone();
+    let guard = state.lock().unwrap();
+
+    let ip_address = match input.into_value(span)? {
+        Value::String { val, .. } => format_ip_address(val),
+        Value::Nothing { .. } => {
+            if let Some(address) = call.opt(engine_state, stack, 0)? {
+                format_ip_address(address)
+            } else {
+                return Err(ShellError::GenericError {
+                    error: "No IP address provided".to_string(),
+                    msg: "".to_string(),
+                    span: None,
+                    help: Some("Provide IP as positional parameter or piped input".into()),
+                    inner: vec![],
+                });
+            }
+        }
+        _ => {
+            return Err(ShellError::GenericError {
+                error: "IP address must be a string".to_string(),
+                msg: "".to_string(),
+                span: None,
+                help: None,
+                inner: vec![],
+            })
+        }
+    };
+
+    let active_cluster = match guard.active_cluster() {
+        Some(c) => c,
+        None => {
+            return Err(no_active_cluster_error(span));
+        }
+    };
+    let org = if let Some(cluster_org) = active_cluster.capella_org() {
+        guard.get_capella_org(cluster_org)
+    } else {
+        guard.active_capella_org()
+    }?;
+
+    let client = org.client();
+    let deadline = Instant::now().add(org.timeout());
+
+    let org_id = find_org_id(ctrl_c.clone(), &client, deadline, span)?;
+
+    let project_id = find_project_id(
+        ctrl_c.clone(),
+        guard.active_project().unwrap(),
+        &client,
+        deadline,
+        span,
+        org_id.clone(),
+    )?;
+
+    let json_cluster = cluster_from_conn_str(
+        active_cluster.display_name().unwrap_or("".to_string()),
+        ctrl_c.clone(),
+        active_cluster.hostnames().clone(),
+        &client,
+        deadline,
+        span,
+        org_id.clone(),
+        project_id.clone(),
+    )?;
+
+    client
+        .allow_ip_address(
+            org_id,
+            project_id,
+            json_cluster.id(),
+            ip_address,
+            deadline,
+            ctrl_c,
+        )
+        .map_err(|e| client_error_to_shell_error(e, span))?;
+
+    Ok(PipelineData::empty())
+}
+
+fn format_ip_address(ip_address: String) -> String {
+    if !ip_address.ends_with("/32") {
```

Comment:
ip address can't be any other format can it? Just checking that we aren't blocking some usage.

Reply:
The addresses required a subnet mask, and if not given the UI defaults to try and use '/32', so I've changed this to behave the same.